pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java

```diff
@@ -238,42 +227,67 @@ public int hashCode() {
    * The <code>DataType</code> enum is used to demonstrate the data type of a column.
    * <p>Array <code>DataType</code> is only used in {@link com.linkedin.pinot.common.utils.DataTableBuilder.DataSchema}.
    * <p>In {@link Schema}, use non-array <code>DataType</code> only.
+   * <p>In pinot, we store data using 5 <code>DataType</code>s: INT, LONG, FLOAT, DOUBLE, STRING. All other
+   * <code>DataType</code>s will be converted to one of them.
    */
   public enum DataType {
-    BOOLEAN,
-    BYTE,
-    CHAR,
-    SHORT,
+    BOOLEAN,      // Stored as STRING.
+    BYTE,         // Stored as INT.
+    CHAR,         // Stored as STRING.
+    SHORT,        // Stored as INT.
     INT,
     LONG,
     FLOAT,
     DOUBLE,
     STRING,
-    OBJECT,
+    OBJECT,       // Used in dataTable to transfer data structure.
     //EVERYTHING AFTER THIS MUST BE ARRAY TYPE
-    BYTE_ARRAY,
-    CHAR_ARRAY,
-    SHORT_ARRAY,
+    BYTE_ARRAY,   // Unused.
+    CHAR_ARRAY,   // Unused.
+    SHORT_ARRAY,  // Unused.
     INT_ARRAY,
     LONG_ARRAY,
     FLOAT_ARRAY,
     DOUBLE_ARRAY,
     STRING_ARRAY;
 
     public boolean isNumber() {
-      return (this == BYTE) || (this == SHORT) || (this == INT) || (this == LONG) || (this == FLOAT)
-          || (this == DOUBLE);
+      return (this == INT) || (this == LONG) || (this == FLOAT) || (this == DOUBLE);
```

Comment:
is there any performance impact of original method ? 

If not, it's better to leave original method as is even if short is stored as int.